Components/FlightControl/HDITask.cpp

```diff
@@ -0,0 +1,172 @@
+/**
+******************************************************************************
+* File Name          : HDITask.cpp
+* Description        : Primary flight task, default task for the system.
+******************************************************************************
+*/
+#include "HDITask.hpp"
+#include "GPIO.hpp"
+#include "SystemDefines.hpp"
+#include "RocketSM.hpp"
+#include "FlightTask.hpp"
+#include <map>
+#include "Command.hpp"
+
+extern TIM_HandleTypeDef htim2;
+
+std::map <RocketState, BLINK> stateBlinks = {
+    {RS_PRELAUNCH, {2, 1000}},
+    {RS_FILL,{3, 1000}},
+    {RS_ARM, {4, 1000}},
+    {RS_IGNITION, {5, 1000}},
+    {RS_LAUNCH, {6, 1000}},
+    {RS_BURN, {7, 1000}},
+    {RS_COAST, {8, 1000}},
+    {RS_DESCENT,{9, 1000}},
+    {RS_RECOVERY, {10, 1000}},
+    {RS_ABORT,{1, 1000}}
+};
+
+/**
+* @brief Constructor for HDITask
+*/
+HDITask::HDITask():Task(HDI_TASK_QUEUE_DEPTH_OBJS)
+{
+}
+
+
+
+/**
+* @brief Initialize the HDITask
+*/
+void HDITask::InitTask()
+{
+    // Make sure the task is not already initialized
+    SOAR_ASSERT(rtTaskHandle == nullptr, "Cannot initialize flight task twice");
+
+
+
+    BaseType_t rtValue =
+        xTaskCreate((TaskFunction_t)HDITask::RunTask,
+            (const char*)"HDITask",
+            (uint16_t)HDI_TASK_STACK_DEPTH_WORDS,
+            (void*)this,
+            (UBaseType_t)HDI_TASK_RTOS_PRIORITY,
+            (TaskHandle_t*)&rtTaskHandle);
+
+
+
+    SOAR_ASSERT(rtValue == pdPASS, "HDITask::InitTask() - xTaskCreate() failed");
+}
+
+
+
+/**
+* @brief Instance Run loop for the Flight Task, runs on scheduler start as long as the task is initialized.
+* @param pvParams RTOS Passed void parameters, contains a pointer to the object instance, should not be used
+*/
+void HDITask::Run(void * pvParams)
+{
+	while (1) {
+	        Command cm;
+
+	        //Wait forever for a command
+	        qEvtQueue->ReceiveWait(cm);
+
+	        //Process the command
+	        HandleCommand(cm);
+	    }
```

Comment:
Keep in mind we might want this buzz / blink to loop constantly iterate instead of just doing this on state transitions. But this may not be needed.